modules/signals/spec/signal-store-feature.spec.ts

```diff
@@ -0,0 +1,94 @@
+import { Signal, signal } from '@angular/core';
+import {
+  selectSignal,
+  signalStore,
+  signalStoreFeature,
+  type,
+  withMethods,
+  withSignals,
+  withState,
+} from '../src';
+import { STATE_SIGNAL } from '../src/signal-state';
+
+describe('signalStoreFeature', () => {
+  function withCustomFeature1() {
+    return signalStoreFeature(
+      withState({ foo: 'foo' }),
+      withSignals(({ foo }) => ({ bar: selectSignal(() => foo() + 1) })),
+      withMethods(({ foo, bar }) => ({
+        baz: () => foo() + bar() + 2,
+      }))
+    );
+  }
+
+  function withCustomFeature2() {
+    return signalStoreFeature(
+      withCustomFeature1(),
+      withMethods(({ foo, baz }) => ({
+        bar: (value: number) => value,
+        m: () => foo() + baz() + 3,
+      }))
+    );
+  }
+
+  function withCustomFeatureWithInput<_>() {
```

Comment:
Does this generic has a meaning?

```suggestion
  function withCustomFeatureWithInput() {
```

Reply:
I noticed a weird typing bug when custom features with input are used in complex use cases.

[Here is the reproduction.](https://github.com/markostanimirovic/ngrx-signal-store-playground/blob/main/src/tests/test2.ts#L20) If we remove `<_>` from `withY`, the compilation error will be thrown in SignalStores that use `withY` feature although the generic isn't used anywhere. 👀

This is a rare case, but I'll document this behavior and there will be a recommendation in the docs.